opensearch-operator/pkg/builders/cluster.go

```diff
@@ -358,13 +358,14 @@ func NewSTSForNodePool(
 
 	var initContainers []corev1.Container
 	if !helpers.SkipInitContainer() {
+		uid, gid := helpers.ResolveUidGid(cr)
 		initContainers = append(initContainers, corev1.Container{
 			Name:            "init",
 			Image:           initHelperImage.GetImage(),
 			ImagePullPolicy: initHelperImage.GetImagePullPolicy(),
 			Resources:       resources,
 			Command:         []string{"sh", "-c"},
-			Args:            []string{"chown -R 1000:1000 /usr/share/opensearch/data"},
```

Comment:
The `1000:1000` is based on official OpenSearch docker image https://github.com/opensearch-project/opensearch-build/blob/main/docker/release/dockerfiles/opensearch.al2023.dockerfile#L19-L20.